### ISSUE TITLE
App crashes if config/package does not contain certain values

### ISSUE THREAD

**Issue by NervosaX** [fyodorvi/jspm-watch#4]
Part of my package.json: if some of these don't exist, the watcher complains that the value doesn't exist. I had to read the source code to figure out where this was. 

```
  "jspm": {
    "configFile": "jspm.config.js", // Required
    "directories": {
      "baseURL": ".", // Required
      "packages": "jspm_packages" // Required
    }
  }
```

It was here:

https://github.com/fyodorvi/jspm-watch/blob/master/index.js#L380-L382

---

Also my config had to have a baseURL, or it failed here:

https://github.com/fyodorvi/jspm-watch/blob/master/index.js#L400

**Comment by AckerApple** [comment]
If I get a chance to make a pull request, I suggest index.js Line 369, becomes:
`pjson.directories.baseURL = pjson.directories.baseURL || '.'`
This would fix this issue. I keep needing to infrequently, re-add the baseURL directory definition.

**Comment by AckerApple** [comment]
I use this package. I see it's been months since an update. If I fix this issue and make a pull request, are chances likely we will see an update? Otherwise, plan is to fork and I'd rather not. This issue has become enough of a pain to act on it here within a weeks time. Thank you kindly

**Comment by NervosaX** [comment]
Would also be interested @AckerApple.

**Comment by AckerApple** [comment]
Understood but I don't see evidence that this package is being touched. I would just like confirmation that @fyodorvi is an active user. This users github activity is fairly low with large gaps. EITHER way, this will be fixed but it's to be decided if I will fork OR send a pull request (active-user=pull-request . stale-user=fork)

**Comment by fyodorvi** [comment]
@AckerApple I'm an active user, without any spare time to maintain the library, unfortunately.

**Comment by NervosaX** [comment]
@fyodorvi I imagine you will accept PRs at least?

**Comment by fyodorvi** [comment]
@NervosaX I guess I should publish the beta instead?

**Comment by AckerApple** [comment]
I hope to issue a pull request this week. Thank you kindly for your reply and hard work.

**Comment by NervosaX** [comment]
@fyodorvi Yeah, that's pretty much what I'm running on anyway.

**Comment by AckerApple** [comment]
I am working on this now. Will fork and then issue pull request.

**Comment by AckerApple** [comment]
I will fork the beta version, glad it was mentioned.

**Comment by AckerApple** [comment]
I have done all the work as promised. If my pull request is accepted (https://github.com/fyodorvi/jspm-watch/pull/8), I will certainly be willing to donate more time to enhance the docs and such to make this package an overall more robustly documented package. Thank you kindly.

**Comment by AckerApple** [comment]
I know it's only been 3 days. Just really hoping to see action on this in the not too distant future. Please at least indicate if in review or something to that effect.

**Comment by fyodorvi** [comment]
@AckerApple done, sorry for long waiting

**Comment by AckerApple** [comment]
That was not a long wait at all. These things deserve time to be accurate. Great Job Team!

@NervosaX, this pulled code satisfies all rebuild issues I was having. Please test for yourself since 1.0.0 is now available. Your issue reports more Config settings that I have found requiring reconfiguration. Let us know or close this issue

**Comment by AckerApple** [comment]
@NervosaX, I see you define the Config path as being jspm.config.js . My config file is always named config.js and I've had no problems. I don't believe this pull request addresses config file pathing if it's not config.js .... Dunno, just educated guessing

**Comment by NervosaX** [comment]
@AckerApple Do you run 0.16 or 0.17? 0.17 has jspm.config.js and jspm.browser.js as being the standard name.

I'll go ahead and test the above today.

**Comment by AckerApple** [comment]
I'm on the npm release, 0.16.44 ... You most likely are right about this. Let me know results of testing

**Comment by NervosaX** [comment]
Sorry, will be slow on testing. Too many deadlines right now. :)

**Comment by AckerApple** [comment]
Heyyyyyyyyyyyyyyyyy, so I've moved onto using jspm 0.17

And it includes watch functionality built right in for both the build(was bundle-sfx) and bundle steps. The included watcher functionality does not appear to rebuild everything, just the changes.

Effectively, jspm 0.17 and its --watch command, have rendered jspm-watch near dead to me. I don't see me using this package any further. In fact, if this was my package(jspm-watch), I would point everyone to 0.17 and it's documentation about --watch as seen here: http://jspm.io/0.17-beta-guide/development-bundling.html

**Comment by NervosaX** [comment]
Yeah, the only issue with that is there's no api support for it. I want to run it in a build along side browser-sync, etc.
